content/upcoming/id2vec.md

```diff
@@ -0,0 +1,598 @@
+| 1 | 3 | 3 | 3 | 3 | 0 | 4 | 4 | 3 | 3 | 3 | 3 | 3 | 3 | 3 |
+| 1 | 3 | 3 | 3 | 3 | 4 | 0 | 4 | 3 | 3 | 3 | 3 | 3 | 3 | 3 |
+| 1 | 3 | 3 | 3 | 3 | 4 | 4 | 0 | 3 | 3 | 3 | 3 | 3 | 3 | 3 |
+| 1 | 4 | 4 | 3 | 3 | 3 | 3 | 3 | 0 | 5 | 5 | 5 | 4 | 4 | 4 |
+| 1 | 4 | 4 | 3 | 3 | 3 | 3 | 3 | 5 | 0 | 6 | 6 | 5 | 5 | 5 |
+| 1 | 4 | 4 | 3 | 3 | 3 | 3 | 3 | 5 | 6 | 0 | 6 | 5 | 5 | 5 |
+| 1 | 4 | 4 | 3 | 3 | 3 | 3 | 3 | 5 | 6 | 6 | 0 | 5 | 5 | 5 |
+| 1 | 4 | 4 | 3 | 3 | 3 | 3 | 3 | 4 | 5 | 5 | 5 | 0 | 5 | 5 |
+| 1 | 4 | 4 | 3 | 3 | 3 | 3 | 3 | 4 | 5 | 5 | 5 | 5 | 0 | 5 |
+| 1 | 4 | 4 | 3 | 3 | 3 | 3 | 3 | 4 | 5 | 5 | 5 | 5 | 5 | 0 |
+
+The biggest co-occurrence value is between "app" and "config" - 6 contexts.
+
+The described scheme can be distributed and run on top of Spark or any other solution. Our Data Retrieval engineer
+Alex has a working proof of concept [swivel-spark-prep](https://github.com/src-d/swivel-spark-prep).
+Thus our identifier embedding approach is the explicit factorization. We apply Swivel subsequently, it takes less than 6 hours to train using 4 NVIDIA 1080Ti grade GPUs on a test 1 million x 1 million matrix produced from ⪆100k most starred repositories on GitHub.
+
+## Results
+
+#### Metasyntactic variables
+
+[Metasyntactic variables](https://en.wikipedia.org/wiki/Metasyntactic_variable) are variable names
+which are used as dummy placeholders, e.g. when you demonstrate a coding concept. Most wellknown
+are probably "foo" and "bar". Here are some selected tokens nearest to "foo":
+
+```
+bound, wibble, quux, overloadargs, baz, testing
+mfoo, ifoo, dfoo, myfoo, dofoo, afoo
+```
+
+and "bar":
+
+```
+jprogress, cescroll, rid, jstool, panel, sherlock, drawing
+pbar, mbars, mybar, mpbar, abar, tobar, bar
+```
+
+And yeah! "bar" is to "foo" as "baz" is to "qux" - we've got the proof now.
+
+#### Kings and queens
+
+* "boy" is to "king" as "girl" is to "queen".
+However, replacing "boy" with "man" and "girl" with "woman" does not prove this association.
+I explain it with the fact that "man" is a typical acronym from "manual" and lost it's specificity.
+
+* "bug" - "test" + "expect" = "suppress". Pretty much what we all do with `try: ... except: pass`. 
+
+* "query" is to "database" as "tune" is to "settings".
+
+* "send" is to "receive" as "push" is to "pop".
+
+The latter two improve upon the common GloVe embeddings tremendously:
+
+* "query" is to "database" as "tune" is to ["databank", "tuned", "searchable_database", "registry", "soundtrack"].
+
+* "send" is to "receive" as "push" is to ["pushing", "garner", "earn", "get", "attain"].
+
+#### Synonyms
+
+Tokens nearest to "english":
+
+```
+dutch
+italian
+danish
+german
+finnish
+russian
+swedish
+french
+nglish
+```
+
+"send":
+
+```
+receive
+sending
+dosend
+ssend
+recv
+sent
+onsend
+```
+
+["fizzbuzz"](https://en.wikipedia.org/wiki/Fizz_buzz) is near to "divisible" and "isevenly".
+
+#### Languages
+
+Tokens nearest to "negatif", apparently German and few French:
```

Comment:
After looking at it carefully, at least 6 identifiers of this list come from Turkey